app/controllers/automated_tests_controller.rb

```diff
@@ -35,14 +35,34 @@ def student_interface
     @student = current_role
     @grouping = @student.accepted_grouping_for(@assignment.id)
 
-    unless @grouping.nil?
+    if @grouping.present?
       @grouping.refresh_test_tokens
       @authorized = flash_allowance(:notice,
                                     allowance_to(:run_tests?,
                                                  current_role,
                                                  context: { assignment: @assignment, grouping: @grouping })).value
+
+      if @assignment.enable_student_tests
+        last_student_run = @grouping.test_runs.where(role: @grouping.accepted_students).first
+        if last_student_run.nil?
```

Comment:
This check is not necessary, the only logic that's needed is in the else branch. If the student hasn't run any tests, we still need to take into account the current time, as that might be very far past when the `@assignment.token_start_date` is.